mutate/compress.go

```diff
@@ -76,6 +91,15 @@ func (gz gzipCompressor) MediaTypeSuffix() string {
 	return "gzip"
 }
 
+func (gz gzipCompressor) WithOpt(opt CompressorOpt) Compressor {
+	switch val := opt.(type) {
```

Comment:
I'm not sure I like that we ignore invalid arguments here. If we make this act on `*gzipCompressor` we could make it return a bool (or error) and then the caller can decide if they care about whether `WithOpt` actually did anything.

Alternatively, it feels like it should be possible to reimplement all of this compressor stuff with a closure that is passed by the caller (that creates the writer and configures it). But that would require more work...

Reply:
Maybe we could add a `CustomCompressor` just for cases where you need to configure the compression? Idk...